plutus-core/changelog.d/20230216_173134_michael.peyton-jones_plc_versions.md

```diff
@@ -0,0 +1,9 @@
+### Added
+
+- New language version 1.1
+- PIR programs now have a version, which is optionally parsed (defaulting to the latest version)
+
+### Changed
+
+- `defaultVersion` renamed to `latestVersion`
+- `applyProgram` now merges annotations and takes the latest program version
```

Comment:
This may not be the right thing to do if the fun and the arg have different major versions. Can we require that they have the same major version?

Reply:
And otherwise do what? Give up? I think so long as it's documented this is fairly reasonable behaviour? (Note to self: document it)

Reply:
Why not. A quick-and-dirty `error "not allowed"` is better than allowing it and potentially leading to undefined behavior.

Reply:
I'd rather make it return a `Maybe`. I think this could easily get used in production code paths. I'll try that.